Train models

mercysu opened the pull request:
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

yunxiangfu2001 committed:
Draft of ResMLP Block

yunxiangfu2001 committed:
rename file and simplify forward for ResBlock

EE-LiuYunhao committed:
Resolve the naming collision

EE-LiuYunhao committed:
Drafting resmlp configs

LawManTing committed:
Merge pull request #1 from EE-LiuYunhao/fix_env

Add a Quick-Start CONDA yaml

mercysu committed:
Update vit_base_patch16_224_flowers.py

LawManTing committed:
Add files via upload

LawManTing committed:
Add files via upload

LawManTing committed:
Add files via upload

LawManTing committed:
Rename mlp-mixer-base.py to mlp-mixer-base_WDA.py

mercysu committed:
Rename vit_base_patch16_224_flowers.py to vit_base_patch16_224_WDA_flowers.py

LawManTing committed:
Create mlp-mixer-base_SDA.py

mercysu committed:
Create vit_base_patch16_224_WDA_flowers.py

mercysu committed:
Create vit_base_patch16_224_WDA_flowers.py

mercysu committed:
Delete vit_base_patch16_224_WDA_flowers.py

mercysu committed:
Update vit_base_patch16_224_WDA_flowers.py

mercysu committed:
Create resnet18_flowers_bs128_SDA.py

mercysu committed:
Create mlp-mixer-base_SDA.py

mercysu committed:
Create vit_base_patch16_224_SDA_flowers.py

EE-LiuYunhao committed:
Merge branch 'train_models' into ResMLP

mercysu committed:
Delete new_configs directory

mercysu committed:
Create mlp-mixer-base_SDA.py

mercysu committed:
Create mlp-mixer-large-p16_64xb64_in1k.py

mercysu committed:
Create mlp-mixer-base-p16_64xb64_in1k.py

mercysu committed:
Create metafile.yml

mercysu committed:
Create README.md

mercysu committed:
Create mlp-mixer-base_WDA.py

mercysu committed:
Add files via upload

mercysu committed:
Add files via upload

mercysu committed:
Create new

mercysu committed:
Create new

mercysu committed:
Create new

mercysu committed:
Create new

mercysu committed:
Delete new

mercysu committed:
Create new

mercysu committed:
Add files via upload

mercysu committed:
Delete new

mercysu committed:
Add files via upload

mercysu committed:
Delete new

mercysu committed:
Add files via upload

mercysu committed:
Delete new

mercysu committed:
Add files via upload

mercysu committed:
Delete new

EE-LiuYunhao committed:
re-organize the configs

EE-LiuYunhao committed:
Re-orgnazination of the config files

EE-LiuYunhao committed:
append the training config for ResMLP

EE-LiuYunhao committed:
fix the relative path problem

EE-LiuYunhao committed:
adjuct the paths

EE-LiuYunhao committed:
remove the wrong configs for ResMLP

ShoufaChen commented:
Hi,

Thank you for your contribution.
Would you mind giving some brief notes on your contribution?

EE-LiuYunhao committed:
remove the other wrong configs

cogito233 committed:
add vit train config

cogito233 committed:
add the trainning logs

cogito233 committed:
fix the CAM code and add some results

yunxiangfu2001 committed:
Completed ResMLP

Built ResMLP backbone with appropriate configs.

EE-LiuYunhao committed:
merge different resmlp configs

yunxiangfu2001 committed:
Add ResMLP intro

EE-LiuYunhao committed:
Merge pull request #4 from EE-LiuYunhao/ResMLP

ResMLP

EE-LiuYunhao committed:
Merge branch 'train_models' into pertrain_eval

EE-LiuYunhao committed:
Merge pull request #3 from EE-LiuYunhao/pertrain_eval

Pertrain eval

EE-LiuYunhao committed:
remove the v2 schedular

EE-LiuYunhao committed:
add checkpoint load for ViT-WDA

LawManTing committed:
Update mlp_mixer_sda.py

LawManTing committed:
Update mlp_mixer_wda.py

EE-LiuYunhao committed:
add lamb schedule config for flowers

yunxiangfu2001 committed:
organized resmlp variants

add embedding 768 configs and organize config files

yunxiangfu2001 committed:
add resnet18 and lamb configs

yunxiangfu2001 committed:
lr = 0.001

yunxiangfu2001 committed:
rewrite resmlp

Make resmlp block easier to read and change some parts to make it EXACTLY the same as original implementation.
Changes:
(1) weight initialization from xavier_uniform_ to trunc_normal
(2) share dropout_layer

Note there is almost no change in accuracy

yunxiangfu2001 committed:
Merge branch 'train_models' of https://github.com/EE-LiuYunhao/COMP3340_Transformer_MLP into train_models

EE-LiuYunhao committed:
add the ann_file config

EE-LiuYunhao committed:
enable pretraining

EE-LiuYunhao committed:
enable training on multiple GPUs

EE-LiuYunhao committed:
remove the result images

EE-LiuYunhao commented:
> Hi,
> 
> Thank you for your contribution. Would you mind giving some brief notes on your contribution?

Sorry for the mistake. One of our groupmate launched the PR to the codebase wrongly. She aimed to approve my PR but both she and I failed to realize that what she had actually done was starting a new PR instead of approving the existing one. 

I did not notice the existence of this PR until I tried to delete a merged stale branch in my forked repository. That is totally my bad. 

Again, apologize for our mistakes. I am preparing a new PR to introduce our modifications to you. Could you please just close the PR? Thank you!
